tracer/src/Datadog.Trace/Iast/Aspects/MailKit/MailkitAspect.cs

```diff
@@ -0,0 +1,111 @@
+    /// </summary>
+    /// <param name="instance">the email body instance</param>
+    /// <param name="encoding">the email encoding</param>
+    /// <param name="bodyText">the email text</param>
+    [AspectMethodReplace("MimeKit.TextPart::SetText(System.String,System.String)")]
+    public static void SetText(object instance, string encoding, string bodyText)
+#pragma warning disable DD0005
+    {
+        IMimeKitTextPart? textPart = ConvertToMimekit(instance);
+
+        if (textPart is not null)
+        {
+            textPart.SetText(encoding, bodyText);
+            CheckForVulnerability(bodyText, textPart);
+        }
+    }
+#pragma warning restore DD0005
+
+    /// <summary>
+    /// Launches a email html injection vulnerability if the email body is tainted, it's not escaped and the email is html compatible.
+    /// </summary>
+    /// <param name="instance">the email body</param>
+    /// <param name="encoding">the email ending</param>
+    /// <param name="bodyText">the email text</param>
+    [AspectMethodReplace("MimeKit.TextPart::SetText(System.Text.Encoding,System.String)")]
+    public static void SetTextSystemTextEncoding(object instance, Encoding encoding, string bodyText)
+#pragma warning disable DD0005
+    {
+        IMimeKitTextPart? textPart = ConvertToMimekit(instance);
+
+        if (textPart is not null)
+        {
+            textPart.SetText(encoding, bodyText);
+            CheckForVulnerability(bodyText, textPart);
+        }
+    }
+#pragma warning restore DD0005
+
+    /// <summary>
+    /// Launches a email html injection vulnerability if the email body is tainted, it's not escaped and the email is html compatible.
+    /// </summary>
+    /// <param name="instance">the email body</param>
+    /// <param name="bodyText">the email text</param>
+    [AspectMethodReplace("MimeKit.TextPart::set_Text(System.String)")]
+    public static void SetTextProperty(object instance, string bodyText)
+#pragma warning disable DD0005
+    {
+        IMimeKitTextPart? textPart = ConvertToMimekit(instance);
+
+        if (textPart is not null)
+        {
+            textPart.Text = bodyText;
+            CheckForVulnerability(bodyText, textPart);
+        }
+    }
+#pragma warning restore DD0005
+
+    private static void CheckForVulnerability(string bodyText, IMimeKitTextPart textPart)
+    {
+        try
+        {
+            if (textPart.IsHtml)
+            {
+                IastModule.OnEmailHtmlInjection(bodyText);
+            }
+        }
+        catch (Exception ex)
+        {
+            IastModule.LogAspectException(ex, $"{nameof(MailkitAspect)}.{nameof(CheckForVulnerability)}");
+        }
+    }
+
+    private static IMimeKitTextPart? ConvertToMimekit(object instance)
+    {
+        IMimeKitTextPart? textPart = null;
+        try
+        {
+            textPart = instance.DuckCast<IMimeKitTextPart>();
```

Comment:
should we use `TryDuckCast` instead?

Reply:
We are following the same pattern in other aspects such as JsonDocumentAspects or JavaScriptSerializerAspects. I guess that we should probably log an DuckCast Failure/exception as error. WDYT?

Reply:
oh ok, makes sense to log, I just thought `TryDuckCast` might have been more performant, and then testing the bool result to log or not :idk

Reply:
it says:
>If you know that your proxy is correct and that it will sometimes fail (due to the nature of the integration) you should use TryDuckCast<T> over DuckAs<T> to safely try the proxying and to handle failure. If you don't expect the proxying to fail (i.e. you're not accounting for explicitly known scenarios) then favour DuckCast<T>

so I don't know if you expect the proxy to fail, maybe not so it's ok like this 🤔